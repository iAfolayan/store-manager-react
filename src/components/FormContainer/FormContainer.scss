$white: #fff;
$overlay: #12121275;
$white: #fff;

.wrapper-body {
  display: flex;
  align-items: center;
  justify-content: center;

  padding: {
    top: 40px;
    bottom: 40px;
  }

  height: 100vh;
}

.wrapper {
  background: $overlay;
  color: $white;
  padding: 40px;
	margin: 0 auto;
	border-radius: 0.313rem;
  padding-bottom: 1.563rem;
  text-align: center;
}

.app-name {
  font: size 2rem;
  color: $white;
  padding-bottom: 10px;
}

// Small devices (landscape phones, 576px and up)
@media (max-width: 576px) {
  .wrapper {
    width: 85%;
  }
  .app-name {
    font-size: 1rem;
  }
 }

// Medium devices (tablets, 768px and up)
@media (min-width: 768px) {
  .wrapper {
    width: 85%;
  }
 }

// Large devices (desktops, 992px and up)
@media (min-width: 992px) {
  .wrapper {
    width: 40%;
  }
 }

// Extra large devices (large desktops, 1200px and up)
@media (min-width: 1200px) {
  .wrapper {
    width: 25%;
  }
 }
 