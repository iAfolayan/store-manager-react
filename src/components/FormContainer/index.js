import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import './FormContainer.scss';
import Logo from './images/logo.png';


class FormContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { title, children } = this.props;
    return (
      <div className="wrapper-body d-flex">
        <div className="wrapper">
          <img src={Logo} alt="Logo" />
          <h1 className="app-name">{ title }</h1>
          <Fragment>{children}</Fragment>
        </div>
      </div>
    );
  }
}

FormContainer.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string
};

FormContainer.defaultProps = {
  children: null,
  title: ''
}

export default FormContainer;
