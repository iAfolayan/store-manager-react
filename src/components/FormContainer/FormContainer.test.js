import React from 'react';
import { shallow } from 'enzyme';
import FormContainer from '.';

describe('<FormContainer />', () => {
  test('should rendered Form container without crashing', () => {
    const wrapper = shallow(<FormContainer title="Edit" />);
    expect(wrapper).toMatchSnapshot();
  });
});
