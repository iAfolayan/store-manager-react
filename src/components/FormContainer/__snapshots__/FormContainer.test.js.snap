// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormContainer /> should rendered Form container without crashing 1`] = `
<div
  className="wrapper-body d-flex"
>
  <div
    className="wrapper"
  >
    <img
      alt="Logo"
      src="test-file-stub"
    />
    <h1
      className="app-name"
    >
      Edit
    </h1>
  </div>
</div>
`;
